service/src/main/java/com/fecklessweasel/service/datatier/FileMetadataTable.java

```diff
@@ -0,0 +1,209 @@
+          ResultSet result = insertStatement.getGeneratedKeys();
+          result.next();
+
+          int fid = result.getInt(1);
+
+          insertStatement.close();
+
+          return fid;
+        } catch (SQLException ex) {
+          throw new ServiceException(ServiceStatus.DATABASE_ERROR, ex);
+        }
+    }
+    /**
+     * Looks up the file using its unique identifier
+     *@param connection Connection to the mySQL database
+     *@param fileId The files unique identifier
+     *@return file A ResultSet containing the file tuple - only a single row.
+     */
+    public static ResultSet lookUpFile(Connection connection, int fileId)
+      throws ServiceException{
+
+        CodeContract.assertNotNull(connection, "connection");
+
+        try{
+          PreparedStatement lookUpFileStatement = connection.prepareStatement(LOOKUP_FILE_QUERY);
+          lookUpFileStatement.setInt(1, fileId);
+
+          return lookUpFileStatement.executeQuery();
+        } catch (SQLException ex) {
+            throw new ServiceException(ServiceStatus.DATABASE_ERROR, ex);
+        }
+    }
+    /**
+     * Looks up all files belonging to a certain user
+     *@param connection Connection to the mySQL database
+     *@param user: The username whose files we want to return
+     *@return files A ResultSet containing all file tuples associated with the username
+     */
+    public static ResultSet lookUpUsersFiles(Connection connection, String user)
+      throws ServiceException{
+
+        CodeContract.assertNotNull(connection, "connection");
+        CodeContract.assertNotNullOrEmptyOrWhitespace(user, "user");
+
+        try{
+          PreparedStatement lookUpFileStatement = connection.prepareStatement(LOOKUP_FILES_FROM_USER_QUERY);
+          lookUpFileStatement.setString(1, user);
+
+          return lookUpFileStatement.executeQuery();
+        } catch (SQLException ex) {
+            throw new ServiceException(ServiceStatus.DATABASE_ERROR, ex);
+        }
+
+    }
+
+    /**
+     * Looks up all files belonging to a certain user
+     *@param connection Connection to the mySQL database
+     *@param course The course we want files from
+     *@param university The university that the course belongs to
+     *@result files A ResultSet containing all file tuples associated with the username
+     */
+    public static ResultSet lookUpCourseFiles(Connection connection, String course, String university)
+      throws ServiceException{
+
+              CodeContract.assertNotNull(connection, "connection");
+              CodeContract.assertNotNullOrEmptyOrWhitespace(course, "course");
+              CodeContract.assertNotNullOrEmptyOrWhitespace(university, "university");
+
+              try{
+                PreparedStatement lookUpFileStatement = connection.prepareStatement(LOOKUP_FILES_FROM_USER_QUERY);
+                lookUpFileStatement.setString(1, course);
+                lookUpFileStatement.setString(1, university);
```

Comment:
Two 1's, this code would not work. To test your queries, you should try running them by calling the functions in the ApiRoot controller and then sending a request to localhost:8080/api and making sure that you don't get an exception back.